### PR TITLE
feat: add scrollToSlice for element

### PR DESCRIPTION
**Pull request by sedghi**

### CODE REVIEW

File: packages/tools/src/utilities/viewport/scrollToSlice.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+/**
+ * It uses the metadata to scroll to the slice that is intended. Currently,
+ * only supports imageIdIndex in the metadata, but could be extended to
+ * support other types of metadata such as camera position, focal point, etc.
+ * for volume viewports.
+ * @param element - the HTML Div element scrolling inside
+ * @param metadata - the metadata used for scrolling
+ * @returns Promise that resolves to ImageIdIndex
+ */
+function scrollToSlice(
```
Comment (JamesAPetts) [inline packages/tools/src/utilities/viewport/scrollToSlice.ts:36]: jumpToSlice?
jumpToPosition?

File: packages/tools/src/utilities/viewport/scrollToSlice.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+   * The FrameOfReferenceUID
+   */
+  FrameOfReferenceUID: string;
+};
+
+/**
+ * It uses the metadata to scroll to the slice that is intended. Currently,
+ * only supports imageIdIndex in the metadata, but could be extended to
+ * support other types of metadata such as camera position, focal point, etc.
+ * for volume viewports.
```
Comment (JamesAPetts) [inline packages/tools/src/utilities/viewport/scrollToSlice.ts:31]: I really don't like docs that talk about the future, as they often never get updated.

Instead can you describe what it does _now_, i.e. jumps to imageIdIndex for an element enabled as a stack viewport?

You can write the intention of the future in @privateRemarks.

File: packages/tools/src/utilities/viewport/scrollToSlice.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ * @param element - the HTML Div element scrolling inside
+ * @param metadata - the metadata used for scrolling
+ * @returns Promise that resolves to ImageIdIndex
+ */
+function scrollToSlice(
+  element: HTMLDivElement,
+  metadata: Metadata
```
Comment (JamesAPetts) [inline packages/tools/src/utilities/viewport/scrollToSlice.ts:38]: Metadata is hugely overloaded term, can we call this something like options, with a type of `ScrollToSliceOptions`/`JumpToSliceOptions` or something?

File: packages/tools/src/utilities/viewport/scrollToSlice.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import { Types, getEnabledElement, StackViewport } from '@cornerstonejs/core';
+
+type Metadata = {
```
Comment (JamesAPetts) [inline packages/tools/src/utilities/viewport/scrollToSlice.ts:3]: The type needs to be exposed.

Comment (JamesAPetts) [reply packages/tools/src/utilities/viewport/scrollToSlice.ts:3]: Can we only have the actually used properties on here for now? We can add more when they are actually used.

**Review by JamesAPetts — CHANGES_REQUESTED** [review]
Minor changes requested for clarity/easy of API use.

**Comment by sedghi** [comment]
jumpToAnnotationUID

**Review by JamesAPetts — APPROVED** [review]
LGTM

**Pull request merged**